channel.go

```diff
@@ -1,14 +1,33 @@
 package thingfulx
 
-import (
-	"time"
-)
+import "time"
 
 // Channel represents an individual data channel being published by a Thing out
 // there in the world somewhere.
 type Channel struct {
-	ID         string    `json:"id"`
-	Value      string    `json:"value"`
+	ID           string            `json:"id"`
+	Metadata     []ChannelMetadata `json:"metadata"`
+	Observations []Observation     `json:"observations"`
+}
+
+// ChannelMetadata is an object which contains any metadata required to
+// semantically descibe a measurement collected by a Thing.
+type ChannelMetadata struct {
+	Prop string `json:"property"`
```

Comment:
@mdevilliers - what would you think about adding a type for `Prop` and `Val`; something like:

```go
type NSEntity struct {
  Namespace string
  Value     string
}

type ChannelMetadata struct {
  Prop NSEntity
  Val  NSEntity
}
```

the name is horrible obviously. These would equate to:

```go
var m := ChannelMetadata{
  Prop: NSEntity{Namespace: "m3-litle", Value: "foo"},
  Val: NSEntity{Namespace: "", Value: "bar"},
}
```

What were you trying to achieve by separating out the namespace or schema from the values? Would the above go in that same direction?